The homework for week five is exercises 6 through 10 on pages 86 and 87.

# Exercise 6

6. The PlantGrowth data set contains three different groups, with each representing various plant food diets (you may need to type data(PlantGrowth) to activate it). The group labeled “ctrl” is the control group, while “trt1” and “trt2” are different types of experimental treatment. As a reminder, this subsetting statement accesses the weight data for the control group:
PlantGrowth$weight[PlantGrowth$group==”ctrl”]
and this subsetting statement accesses the weight data for treatment group 1:
PlantGrowth$weight[PlantGrowth$group==”trt1”]
Run a t‐test to compare the means of the control group (“ctrl”) and treatment group 1 (“trt1”) in the PlantGrowth data. Report the observed value of t, the degrees of freedom, and the p‐value associated with the observed value. Assuming an alpha threshold of .05, decide whether you should reject the null hypothesis or fail to reject the null hypothesis. In addition, report the upper and lower bound of the confidence interval.

In [2]:
df <- PlantGrowth

In [5]:
str(df)

'data.frame':	30 obs. of  2 variables:
 $ weight: num  4.17 5.58 5.18 6.11 4.5 4.61 5.17 4.53 5.33 5.14 ...
 $ group : Factor w/ 3 levels "ctrl","trt1",..: 1 1 1 1 1 1 1 1 1 1 ...


In [3]:
head(df)

weight group
1 4.17   ctrl 
2 5.58   ctrl 
3 5.18   ctrl 
4 6.11   ctrl 
5 4.50   ctrl 
6 4.61   ctrl

In [6]:
t.test(df$weight[ df$group == 'ctrl' ], df$weight[ df$group == 'trt1'])


	Welch Two Sample t-test

data:  df$weight[df$group == "ctrl"] and df$weight[df$group == "trt1"]
t = 1.1913, df = 16.524, p-value = 0.2504
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -0.2875162  1.0295162
sample estimates:
mean of x mean of y 
    5.032     4.661 


The t value is 1.1913. The degree of freedom is 16.524. The p-value is 0.2504. Since the alpha value is .05, then we would fail to reject the null hypothesis because alpha is smaller than the p-value.

The confidence interval is (-.2875, 1.0295) where -.2875 is the lower bound and 1.0295 is the upper bound.

# Exercise 7

7. Install and library() the BEST package. Note that you may need to install a program called JAGS onto your computer before you try to install the BEST package inside of R. Use BESTmcmc() to compare the PlantGrowth control group (“ctrl”) to treatment group 1 (“trt1”). Plot the result and document the boundary values that BESTmcmc() calculated for the HDI. Write a brief definition of the meaning of the HDI and interpret the results from this comparison.

In [8]:
#install.packages('BEST')
library('BEST')

also installing the dependency ‘rjags’


Warning message in install.packages("BEST"):
“installation of package ‘rjags’ had non-zero exit status”
Warning message in install.packages("BEST"):
“installation of package ‘BEST’ had non-zero exit status”
Updating HTML index of packages in '.Library'

Making 'packages.html' ...
 done



In [ ]:
plantBest <- BESTmcmc(df$weight[ df$group == 'ctrl' ], df$weight[ df$group == 'trt1'])
plot(plantBest)

The Highest Density Interval (HDI) lets us know which values in the distribution are the most likely, i.e., which values show up the most in the distribution.

The lower and upper limits from the HDI are -0.365 and 1.13 repectively. This gives us a similar result as the t-test where we do not reject the null. This means that we do not have enough information to conclude that the means in both distributions are different from each other.

# Exercise 8

8. Compare and contrast the results of Exercise 6 and Exercise 7. You have three types of evidence: the results of the null hypothesis test, the confidence interval, and the HDI from the BESTmcmc() procedure. Each one adds something, in turn, to the understanding of the difference between groups. Explain what information each test provides about the comparison of the control group (“ctrl”) and the treatment group 1 (“trt1”).

The t-test and the HDI gave us similar results where we cannot conclude any differene between the means from the two groups in the Plant Growth dataset. On the other hand, the most likely values are different from each other. In the HDI, the mean is .388 while in the CDI it is .371. Therefore, they are centered around different numbers. Another difference is the range of the intervals. The HDI is wider than the CDI which means it is less certain about the most likely value.

# Exercise 9

9. Using the same PlantGrowth data set, compare the “ctrl” group to the “trt2” group. Use all of the methods described earlier (t‐test, confidence interval, and Bayesian method) and explain all of the results.

In [17]:
t.test(df$weight[ df$group == 'ctrl' ], df$weight[ df$group == 'trt2'])


	Welch Two Sample t-test

data:  df$weight[df$group == "ctrl"] and df$weight[df$group == "trt2"]
t = -2.134, df = 16.786, p-value = 0.0479
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -0.98287213 -0.00512787
sample estimates:
mean of x mean of y 
    5.032     5.526 


In [11]:
plantBest <- BESTmcmc(df$weight[ df$group == 'ctrl' ], df$weight[ df$group == 'trt2'])
plot(plantBest)

ERROR: Error in BESTmcmc(df$weight[df$group == "ctrl"], df$weight[df$group == : could not find function "BESTmcmc"


Comparing the control group against the trt2 group gives very different results in the HDI and the CDI. The HDI's interval contains 0 while the CDI does not. This means that the CDI rejects the null hypothesis while the HDI does not. The t-test also rejects the null hypothesis. The HDI while shifted to the left has a mean of -.485 which means it is leaning to the trt2 having a higher mean, but it is still not confident enough to say there is a difference between the two.

# Exercise 10

10. Consider this t‐test, which compares two groups of n = 100,000 observations each:
t.test(rnorm(100000,mean=17.1,sd=3.8),rnorm(100000,mean=17.2,sd=3.8))

For each of the groups, the rnorm() command was used to generate a random normal distribution of observations similar to those for the automatic transmission group in the mtcars database (compare the programmed standard deviation for the random normal data to the actual mtcars data). The only difference between the two groups is that in the first rnorm() call, the mean is set to 17.1 mpg and in the second it is set to 17.2 mpg. I think you would agree that this is a negligible difference, if we are discuss‐ ing fuel economy. Run this line of code and comment on the results of the t‐test. What are the implications in terms of using the NHST on very large data sets?

In [12]:
t.test(rnorm(100000,mean=17.1,sd=3.8),rnorm(100000,mean=17.2,sd=3.8))


	Welch Two Sample t-test

data:  rnorm(1e+05, mean = 17.1, sd = 3.8) and rnorm(1e+05, mean = 17.2, sd = 3.8)
t = -6.8011, df = 2e+05, p-value = 1.041e-11
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -0.14917636 -0.08243044
sample estimates:
mean of x mean of y 
  17.0775   17.1933 


The t-test is extremely confident that the two distributions have a different mean. This is partly because having a huge dataset will almost always result in a difference in the mean since the confidence increases as you add more datapoints.